### PR TITLE
Highlight paired brackets

### LINKED ISSUE #104: More obvious highlighting of paired braces
**Describe the current state of the problem**
Right now, a paired bracket is outlined with a gray square, which is poorly visible and works at specific cursor positions.

image

image

**Describe the solution you'd like**
When you put a cursor in the code next to the curly bracket, then this curly bracket and its pair should be highlighted in a more noticeable color.

As seen in VSCode: 
image

**Describe alternatives you've considered (optional)**
A color highlight for all the brackets (aka rainbow brackets in most code editors).

As seen in VSCode & Horizon theme (too messy, I guess?): 
image

### PR DESCRIPTION
**Pull request by qualitymanifest**
Closes #104

Lints and builds fine, and seems to do what it's supposed to. I found a way to grab the `Range` constructor out of ace so I was able to override `highlightBrackets` without touching ace.js.

I tried to keep it current with your develop branch but I missed your last commit by a couple minutes. I also butchered the commit emoji, sorry!

**Ping @CosmoMyzrailGorynych**

### CODE REVIEW

File: src/js/aceHelpers.js [hunk 1]
```diff
@@ -139,6 +139,40 @@
...
+                if (matchPos) {
+                    matchRange = new Range(matchPos.row, matchPos.column, matchPos.row, matchPos.column + 1);
+                }
+            }
+            if (matchRange) {
+                var cursorRange = new Range(cursorPos.row, cursorPos.column - 1, cursorPos.row, cursorPos.column);
+                session.$bracketHighlight = session.addMarker(cursorRange, 'ace_bracket', 'text');
+                session.$bracketHighlightMatch = session.addMarker(matchRange, 'ace_bracket', 'text');
+            }
+        }, 50);
```
Comment (CosmoMyzrailGorynych) [inline src/js/aceHelpers.js]: Does it work with `0` (at L173)?

**Comment by CosmoMyzrailGorynych** [comment]
So I set that timeout to 0, then removed it at all and bracket highlighting still worked, without the tiny flicker on closely put brackets.
Was it something that was left during the development, or does it have some function, @qualitymanifest? I don't like the current `setTimeout(...., 50)`, it gives a sense of ambiguity and of something random 🙃

**Comment by qualitymanifest** [comment]
Cool - if it doesn't cause any issues to remove it, might as well get rid of it. The timeout is something that is leftover from the [original `$highlightBrackets` function](https://github.com/ajaxorg/ace/blob/902d1a02864479cfa7c47f46972ff53fb56924fc/lib/ace/editor.js#L523) from the ace.js source. I started out with that function and then heavily modified it, but I figured the timeout was there for a good reason so I left that.

If the timeout is removed I think it should be okay to remove everything to do with `this.$highlightPending` as well. Do you want me to make these changes?

**Comment by qualitymanifest** [comment]
I tried it out and you're right, it seems to work well without the timer so I removed it.

**Comment by CosmoMyzrailGorynych** [comment]
It is said that [setTimeout was added to prevent browser blocking](https://github.com/ajaxorg/ace/blob/902d1a02864479cfa7c47f46972ff53fb56924fc/lib/ace/editor.js#L533), though it was needed five years ago. Maybe it is not relevant now, after all.

**Review by CosmoMyzrailGorynych — APPROVED** [review]
All is good!

**Pull request merged**